crates/nu-cli/src/completions/file_completions.rs

```diff
@@ -105,6 +105,27 @@ pub fn partial_from(input: &str) -> (String, String) {
     // rsplit_once removes the separator
     base.push(SEP);
 
+    // take leftmost dots from rest and push them into base
+    let mut i = 0;
+
+    while i < rest.len() && rest.chars().nth(i).expect("i is within rest length") == '.' {
+        i += 1;
+    }
+
+    if i > 0 {
+        let (leftmost_dots, rest) = rest.split_at(i);
```

Comment:
This code is not Unicode safe this way. `chars` can be more than 1 byte per character.